crates/oci-distribution/src/lib.rs

```diff
@@ -108,23 +113,26 @@ impl Client {
         }
     }
 
-    /// Pull a manifest from the remote OCI Distribution service.
-    ///
-    /// If the connection has already gone through authentication, this will
-    /// use the bearer token. Otherwise, this will attempt an anonymous pull.
-    pub async fn pull_manifest(&self, image: &Reference) -> OciResult<OciManifest> {
-        let client = reqwest::Client::new();
-        let url = image.to_v2_manifest_url();
-        let request = client.get(&url);
-
+    fn auth_headers(&self) -> HeaderMap {
         let mut headers = HeaderMap::new();
         headers.insert("Accept", "application/vnd.docker.distribution.manifest.v2+json,application/vnd.docker.distribution.manifest.list.v2+json".parse().unwrap());
 
         if let Some(bearer) = self.token.as_ref() {
             headers.insert("Authorization", bearer.bearer_token().parse().unwrap());
         }
+        headers
+    }
 
-        let res = request.headers(headers).send().await?;
+    /// Pull a manifest from the remote OCI Distribution service.
+    ///
+    /// If the connection has already gone through authentication, this will
+    /// use the bearer token. Otherwise, this will attempt an anonymous pull.
+    pub async fn pull_manifest(&self, image: &Reference) -> OciResult<OciManifest> {
+        // We unwrap right now because this try_from literally cannot fail.
```

Comment:
Does this comment still make sense? There's no unwrapping here.

Reply:
Oh, good catch! Removing that.